### PR TITLE
expression: implement vectorized evaluation for `builtinJSONKeysSig`

### PR DESCRIPTION
**Pull request by js00070**
### What problem does this PR solve? 
implement vectorized `builtinJSONKeysSig` mentioned here https://github.com/pingcap/tidb/issues/12104

### What is changed and how it works?
about 10% faster
```
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONKeysSig-VecBuiltinFunc-4              4027            266976 ns/op           91728 B/op       4914 allocs/op
BenchmarkVectorizedBuiltinJSONFunc/builtinJSONKeysSig-NonVecBuiltinFunc-4           4203            290651 ns/op           91728 B/op       4914 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_json_vec.go [hunk 1]
```diff
@@ -27,11 +29,36 @@ func (b *builtinJSONDepthSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
...
+	var j json.BinaryJSON
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+
+		j = buf.GetJSON(i)
+		if j.TypeCode != json.TypeCodeObject {
+			result.AppendNull()
```
Comment (qw4990) [inline expression/builtin_json_vec.go]: Should we return a `json.ErrInvalidJSONData` error in this case?

Comment (js00070) [reply expression/builtin_json_vec.go]: yes, I should.
But I don't know whether or not it should return the error immediately in the `for` loop, or return the error at the end of the function?

Comment (js00070) [reply expression/builtin_json_vec.go]: I also have the same confusion in other vectorized function

Comment (Reminiscent) [reply expression/builtin_json_vec.go]: Normally, if ```err != nil```, we need to return error directly. But there are some cases where the error is reprocessed, it should recheck the ```err```.

**Pull request merged**